test/e2e/functional/tests/e2e.go

```diff
@@ -898,6 +898,51 @@ var _ = Describe("metallb", func() {
 			}, 1*time.Minute, 5*time.Second).Should(BeTrue(), "Failed to delete second AddressPool resource")
 
 		})
+
+		It("Should recognize duplicate BGPAdvertisements in AddressPool", func() {
+			By("Creating AddressPool resource with duplicate BGPAdvertisements")
+			autoAssign := false
+			testAddressPool := &metallbv1beta1.AddressPool{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-addresspool",
+					Namespace: OperatorNameSpace,
+				},
+				Spec: metallbv1beta1.AddressPoolSpec{
+					Protocol: "bgp",
+					Addresses: []string{
+						"1.1.1.1-1.1.1.100",
+					},
+					AutoAssign: &autoAssign,
+					BGPAdvertisements: []metallbv1beta1.BgpAdvertisement{
+						{
+							AggregationLength:   pointer.Int32Ptr(24),
+							AggregationLengthV6: pointer.Int32Ptr(124),
+							LocalPref:           100,
+							Communities: []string{
+								"65535:65282",
+								"7003:007",
+							},
+						},
+						{
+							AggregationLength:   pointer.Int32Ptr(24),
+							AggregationLengthV6: pointer.Int32Ptr(124),
+							LocalPref:           100,
+							Communities: []string{
+								"65535:65282",
+								"7003:007",
+							},
+						},
+					},
+				},
+			}
+
+			By("Expecting AddressPool validation webhook to deny the AddressPool")
+			err := testclient.Client.Create(context.Background(), testAddressPool)
+			Expect(err).ToNot(BeNil())
+			if !strings.Contains(fmt.Sprint(err), "duplicate definition of bgpadvertisement") {
+				Expect(err).ToNot(HaveOccurred())
+			}
+		})
```

Comment:
I thought we agreed on duplicating tests in UT and e2e and having additional test cases in UT only is the preferred way

Reply:
ok, i can remove the E2E test